### PR TITLE
Add Phoenix Keyboard

### PR DESCRIPTION
**Pull request by LSChyi**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/phoenix/phoenix.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+inline void led1_on(void) { led_pin_on(8); };
+inline void led2_on(void) { led_pin_on(5); };
+inline void led3_on(void) { led_pin_on(6); };
+
+inline void led1_off(void) { led_pin_off(8); };
+inline void led2_off(void) { led_pin_off(5); };
+inline void led3_off(void) { led_pin_off(6); };
+
+#define LAYOUT_ergodox(                    \
```
Comment (drashna) [inline keyboards/phoenix/phoenix.h:32]: Could you add the `ergodox_pretty` layout too?

File: keyboards/phoenix/rules.mk [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Options to pass to dfu-util when flashing
+DFU_ARGS = -d 0483:df11 -a 0 -s 0x08000000:leave
+DFU_SUFFIX_ARGS = -v 0483 -p df11
+
+# Build Options
+#   change yes to no to disable
+#
+KEYBOARD_SHARED_EP       = yes
+OPT_DEFS += -DSTM32_DMA_REQUIRED=TRUE
+LINK_TIME_OPTIMIZATION_ENABLE=yes
```
Comment (drashna) [inline keyboards/phoenix/rules.mk:55]: This line should be removed, as it tends to cause issues with ARM/ChibiOS. 

```suggestion
```

Also ... as is, this doesn't actually do anything, anymore.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/phoenix/phoenix.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+void keyboard_pre_init_kb(void) {
+    palSetPadMode(GPIOB, 8, PAL_MODE_OUTPUT_PUSHPULL);
+    palSetPadMode(GPIOB, 5, PAL_MODE_OUTPUT_PUSHPULL);
+    palSetPadMode(GPIOB, 6, PAL_MODE_OUTPUT_PUSHPULL);
+    led1_off();
+    led2_off();
+    led3_off();
+}
```
Comment (drashna) [inline keyboards/phoenix/phoenix.c:43]:
```suggestion
    keyboard_pre_init_user();
}
```

File: keyboards/phoenix/phoenix.c [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    led3_on();
+    wait_ms(50);
+
+    led1_off();
+    wait_ms(50);
+    led2_off();
+    wait_ms(50);
+    led3_off();
+    wait_ms(50);
+}
```
Comment (drashna) [inline keyboards/phoenix/phoenix.c:59]:
```suggestion
    keyboard_post_init_user();
}
```

File: keyboards/phoenix/phoenix.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+
+layer_state_t layer_state_set_kb(uint32_t layer_state) {
+    uint8_t layer = biton(layer_state);
+    manipulate_led(1, layer  & 1);
+    manipulate_led(2, layer >> 1 & 1);
+    manipulate_led(3, layer >> 2 & 1);
+    return layer_state;
+}
```
Comment (drashna) [inline keyboards/phoenix/phoenix.c:79]:
```suggestion
layer_state_t layer_state_set_kb(layer_state_t state) {
    state = layer_state_set_user(state);
    uint8_t layer = get_highest_layer(state);
    manipulate_led(1, layer  & 1);
    manipulate_led(2, layer >> 1 & 1);
    manipulate_led(3, layer >> 2 & 1);
    return state;
}
```

File: keyboards/phoenix/info.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{
+    "keyboard_name": "Phoenix",
+    "url": "ergokb.tw",
+    "maintainer": "ErgoKB via lschyi",
+    "manufacturer": "ErgoKB",
+    "width": 17,
+    "height": 8,
+
+    "layouts": {
```
Comment (drashna) [inline keyboards/phoenix/info.json:9]: Would like to have `ergodox pretty` added here too.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/phoenix/rules.mk [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+BOOTMAGIC_ENABLE = lite       # Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
+CONSOLE_ENABLE = no         # Console for debug(+400)
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
```
Comment (noroadsleft) [inline keyboards/phoenix/rules.mk:13]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
```

File: keyboards/phoenix/rules.mk [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (noroadsleft) [inline keyboards/phoenix/rules.mk:17]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**